### PR TITLE
LLT-5211 Wait for drivers loaded in Windows

### PR DESCRIPTION
**Pull request by matislovas**
### Problem
In Windows, we're sometimes seeing the errors: `Telio::start_named: Err(Adapter(WireguardGo(FailedToStart(-1))))` (the same with WireguardNT adapter). Primary suspect is the `*.dll` loader mechanism, which returns, but still haven't fully initialized the library (error happens only on first library load).

### Solution
Add windows service checker for `Wintun` and `WireGuard` services.

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

File: crates/telio-wg/src/windows/service.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            SERVICE_QUERY_STATUS, SERVICE_STATUS,
+        },
+    },
+};
+
+pub const DEFAULT_SERVICE_WAIT_TIMEOUT: Duration = Duration::from_secs(5);
+
+#[derive(Debug)]
+enum WinServiceStatus {
+    Stopped = 1,
```
Comment (gytsto) [inline crates/telio-wg/src/windows/service.rs:21]: would be nice to get a comment where the values are coming from :smile:

Comment (matislovas) [reply crates/telio-wg/src/windows/service.rs:21]: They're taken from [here](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-service_status). I did not find an enum in `windows-rs` crate, so made one from there.

Comment (gytsto) [reply crates/telio-wg/src/windows/service.rs:21]: isn't this the one -> https://docs.rs/windows-service/latest/windows_service/service/enum.ServiceState.html

Comment (gytsto) [reply crates/telio-wg/src/windows/service.rs:21]: ah but not in `windows-rs`

Comment (matislovas) [reply crates/telio-wg/src/windows/service.rs:21]: Yeah the same, but that's a different crate, and I though siding with our plain implementation

Comment (gytsto) [reply crates/telio-wg/src/windows/service.rs:21]: anyway, good enough :smile:

**Review by LukasPukenis — COMMENTED** [review]
+1. It makes sense what it does, but it 's hard to validate that it does so correctly without a testcase :)

**Comment by matislovas** [comment]
> +1. It makes sense what it does, but it 's hard to validate that it does so correctly without a testcase :)

Yeah, will add it, but as I imagine, it will still be flaky. I can un-load the driver in windows, and wait for successful `tcli` startup. But even if my service monitor will not work, the test won't fail in all cases (will be flaky).

**Comment by matislovas** [comment]
P.S. test is added

**Review by gytsto — APPROVED** [review]
+1

**Pull request merged**